bundles/admin/admin-announcements/publisher/AnnouncementsTool.js

```diff
@@ -65,7 +65,10 @@ Oskari.clazz.define('Oskari.admin.bundle.admin-announcements.publisher.Announcem
                     toolPluginAnnouncementsConf.config.announcements.forEach(announcement => {
                         const filteredAnnouncement = me.announcements.filter(ann => ann.id === announcement);
                         if (me.isAnnouncementValid(filteredAnnouncement[0])) {
-                            me.selectedAnnouncements.push(filteredAnnouncement[0]);
+                            // make sure there are no duplicates
+                            if (me.selectedAnnouncements.filter(ann => ann.id === filteredAnnouncement[0].id).length === 0) {
+                                me.selectedAnnouncements.push(filteredAnnouncement[0]);
+                            }
                         }
```

Comment:
Related to the other suggestion about renaming: 
```suggestion
                            // make sure there are no duplicates
                            if (me.selectedAnnouncements.find(ann => ann.id === announcement.id).length === 0) {
                                me.selectedAnnouncements.push(announcement);
                            }
                        }
```

Reply:
can't switch the filter to find, without adding undefined check, which would make it less readable in my opinion. The renaming is very valid, but I feel the possibility of undefined is more dangerous than filter returning an empty array.

Reply:
The ! at the beginning of my suggestion should be enough since if find doesn't return the announcement it will add it to the selectedAnnouncements. But it's a bit hard to spot at the start and maybe it's best to use array.some() instead to make it even clearer:
```suggestion
                            // make sure there are no duplicates
                            const alreadyAdded = me.selectedAnnouncements.some(ann => ann.id === announcement.id);
                            if (!alreadyAdded) {
                                me.selectedAnnouncements.push(filteredAnnouncement[0]);
                            }
                        }
```

Reply:
But it's not bad. The main thing bugging me was the naming/array usage for `filteredAnnouncement[0]` :)